homeassistant/components/litterrobot/__init__.py

```diff
@@ -24,7 +24,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await hub.login(load_robots=True)
 
     if any(hub.litter_robots()):
-        await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+        platforms = [
+            platform
+            for platform in PLATFORMS
+            if platform not in (Platform.BUTTON, Platform.VACUUM)
+        ]
```

Comment:
Would it be better to set the default PLATFORMS constant to not include BUTTON and VACUUM and then just add them instead with the logic here?

Reply:
I think `PLATFORMS` by convention always includes all platforms the integration can possibly support.

Reply:
You could also make a `dict` to map classes to platforms and do a lookup with a default.

Reply:
Can you point to an example?

Reply:
This might be semi similar: https://github.com/home-assistant/core/blob/dev/homeassistant/components/switchbot/__init__.py#L31

Reply:
Do you have the model names for all 4 supported devices if I am counting correctly?

Reply:
Instead of models, you should just be able to use the classes: `LitterRobot3`, `LitterRobot4`, `FeederRobot`